routing/http/contentrouter/contentrouter_test.go

```diff
@@ -108,26 +128,40 @@ func TestFindProvidersAsync(t *testing.T) {
 
 	p1 := peer.ID("peer1")
 	p2 := peer.ID("peer2")
-	ais := []types.ProviderResponse{
-		&types.ReadBitswapProviderRecord{
-			Protocol: "transport-bitswap",
-			Schema:   types.SchemaBitswap,
-			ID:       &p1,
+	p3 := peer.ID("peer3")
+	p4 := peer.ID("peer4")
+	ais := []types.Record{
+		&types.PeerRecord{
+			Schema:    types.SchemaPeer,
+			ID:        &p1,
+			Protocols: []string{"transport-bitswap"},
 		},
-		&types.ReadBitswapProviderRecord{
-			Protocol: "transport-bitswap",
+		//lint:ignore SA1019 // ignore staticcheck
+		&types.BitswapRecord{
+			//lint:ignore SA1019 // ignore staticcheck
 			Schema:   types.SchemaBitswap,
 			ID:       &p2,
+			Protocol: "transport-bitswap",
+		},
+		&types.PeerRecord{
+			Schema:    types.SchemaPeer,
+			ID:        &p3,
+			Protocols: []string{"transport-bitswap"},
+		},
+		&types.PeerRecord{
+			Schema:    types.SchemaPeer,
+			ID:        &p4,
+			Protocols: []string{"transport-horse"},
 		},
-		&types.UnknownProviderRecord{
-			Protocol: "UNKNOWN",
+		&types.UnknownRecord{
```

Comment:
:warning:  @hacdias mind adding `ID` field here and including it in expects to ensure we are not ignoring responses with `Schema: "unknown"` that have valid `ID` (and optional `Addrs`)?

Rationale: cid.contact returns all nft.storage announcements like this ([example](https://cid.contact/routing/v1/providers/bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi)) so it is really important to not have regressions here:

```json
{
  "Protocol":"unknown",
  "Schema":"unknown",
  "ID":"QmUA9D3H7HeCYsirB3KmPSvZh3dNXMZas6Lwgr4fv1HTTp",
  "Addrs":["/dns4/dag.w3s.link/tcp/443/https"]
}
```

 to ensure smooth transition, the client should detect when `UnknownRecord` has `ID` and `Addrs`  fields and convert it to `PeerRecord`.

Reply:
@lidel they are ignored at the moment too: https://github.com/ipfs/boxo/blob/303595bcdba7c4ff1e4186a8089a550df6da9d28/routing/http/contentrouter/contentrouter.go#L105-L138

Reply:
Also see https://github.com/ipfs/kubo/pull/9877#discussion_r1305482906.